### ISSUE TITLE
Kubelet skipNodeRegistration breaks upgrade-k8s

### ISSUE THREAD

**Issue by maxpain** [siderolabs/talos#9403]
## Bug Report
Kubelet `skipNodeRegistration` breaks `talosctl upgrade-k8s`

### Description
Hello. I want to hide the control plane nodes from the Kubernetes cluster using this configuration:
```yaml
machine:
  kubelet:
    skipNodeRegistration: true
```

But `talosctl upgrade-k8s` stops working:

```sh
% talosctl -n cp1 upgrade-k8s         
error detecting the lowest Kubernetes version failed to detect lowest Kubernetes version
```

### Environment

- Talos version: v1.8.0
- Kubernetes version: 1.31.1
- Platform: Hetzner Cloud

**Comment by smira** [comment]
Yes, it's not supported this way, as `talosctl` won't have a way to verify fully, but you can still update [manually](https://www.talos.dev/v1.8/kubernetes-guides/upgrading-kubernetes/#manual-kubernetes-upgrade).

It should be possible to implement that, but right now we're not looking into that.
